### ISSUE TITLE
[Autoware Labs] Dense Urban ODD - Reviewing and Tracking Failed Scenarios

### ISSUE THREAD

**Issue by brkay54** [autowarefoundation/autoware.universe#7485]
### Checklist

- [X] I've read the [contribution guidelines](https://github.com/autowarefoundation/autoware/blob/main/CONTRIBUTING.md).
- [X] I've searched other issues and no duplicate issues were found.
- [X] I've agreed with the maintainers that I can plan this task.

### Description

[Autoware Labs] Dense Urban ODD - Reviewing and Tracking Failed Scenarios
**Follow-up task of:**
- https://github.com/autowarefoundation/autoware/issues/4636

### Objective
This issue tracks the progress and improvements related to the scenario catalog created for autonomous driving tests within a Dense Urban Operational Design Domain (ODD). The primary objectives are:
- **Defining Problems:** Clearly define the problems for developers using the scenarios below.
- **Tracking Development:** Monitor the development process.

### Dense Urban ODD Characteristics
- **Low-Speed Driving:** Typical speeds around 15 km/h.
- **Complex Traffic Elements:** Includes dense traffic with various dynamic interactions.
- **Dynamic Agents:** Moving pedestrians and cyclists are frequently encountered.
- **Bidirectional Movement:** Vehicles operate in lanes supporting bidirectional flow.
- **Traffic Rule Non-compliance:** Other road users may exhibit behaviors such as:
  - Driving in the wrong direction.
  - Cutting in unexpectedly from off-road areas.
  - Pedestrians jaywalking.
  - Drivers overtaking improperly.

### Task for Developers
- **Investigating failed scenarios:** Identify root causes and potential fixes for failed scenarios.
- **Improving Software Packages:**  Enhance functionalities based on scenario feedback.
  - **Fine-Tuning Parameters:** If the package already has the functionality to handle the failed case, fine-tune the package parameters.
  - **Proposing Improvements:** If the package has the functionality but it is insufficient to handle the failed case, propose improvements to the functionality.
  - **Adding New Functionalities:**  If the package lacks the functionality to handle the failed case, propose adding new functionality to the target package.
- **Creating Additional Scenarios:** Develop new scenarios to test various other conditions and ensure robustness.

### List of the scenarios
<details>
<summary>Click to expand the list of the scenarios</summary>

#### Useful Links

- **Scenario Suit:** [Autoware Evaluator](https://evaluation.ci.tier4.jp/evaluation/suites/05e1818a-85a8-4940-afb1-9e1795fe7b7f?project_id=awf)
- **Scenario Description Document:** [Google Presentation](https://docs.google.com/presentation/d/1t8PaJIXKCQ7GElig4dVbbkcjLEphHQ9ARmWCci5IML4/edit#slide=id.g2d072098d1a_0_4)
<details>
<summary>UC-NTR-001-0001</summary>

- **Scenario Description:** Ego vehicle is overtaken from outside of the road by a faster moving NPC.
- **Development Focus:** Ensure ego vehicle remains stationary until the high velocity NPC has completed its overtaking maneuver.
- **Scenario Links:** 
	- [UC-NTR-001-0001_case1](https://evaluation.ci.tier4.jp/evaluation/scenarios/0c51ed83-bffa-419e-9725-c649fe4780f3?project_id=awf)
	- [UC-NTR-001-0001_case2](https://evaluation.ci.tier4.jp/evaluation/scenarios/70003386-d425-48ed-8746-7a2368d53b67?project_id=awf)
- **Map Link:** [LEO-VM-0001](https://evaluation.ci.tier4.jp/evaluation/maps/972?project_id=awf)
- **Issue/PR Links:**
</details>

<details>
<summary>UC-NTR-001-0002</summary>

- **Scenario Description:** Ego vehicle is overtaken from outside of the road by a slower moving NPC.
- **Development Focus:** Ensure ego vehicle departs before the NPC completes its overtaking maneuver because its velocity is too low.
- **Scenario Links:** 
	- [UC-NTR-001-0002_case1](https://evaluation.ci.tier4.jp/evaluation/scenarios/767d8164-3cf8-4682-9aa3-c19eb796d166?project_id=awf)
	- [UC-NTR-001-0002_case2](https://evaluation.ci.tier4.jp/evaluation/scenarios/180c2e27-e623-4080-8c1e-f9853d475235?project_id=awf)
- **Map Link:** [LEO-VM-0001](https://evaluation.ci.tier4.jp/evaluation/maps/972?project_id=awf)
- **Issue/PR Links:**
</details>

<details>
<summary>UC-NTR-001-0003</summary>

- **Scenario Description:** Ego vehicle is overtaken from the same direction lane by a faster moving NPC before starting.
- **Development Focus:** Ensure ego vehicle remains stationary until the high velocity NPC has completed its overtaking maneuver.
- **Scenario Links:** 
	- [UC-NTR-001-0003_case1](https://evaluation.ci.tier4.jp/evaluation/scenarios/b1bbb34c-a939-46b3-818b-cb8d88dd87d7?project_id=awf)
	- [UC-NTR-001-0003_case2](https://evaluation.ci.tier4.jp/evaluation/scenarios/37a18e54-a640-400f-8b53-aa4f6d2b6772?project_id=awf)
- **Map Link:** [LEO-VM-00009](https://evaluation.ci.tier4.jp/evaluation/maps/980?project_id=awf)
- **Issue/PR Links:**
</details>

<details>
<summary>UC-NTR-001-0004</summary>

- **Scenario Description:** Ego vehicle is overtaken from the same direction lane by a slower moving NPC before starting.
- **Development Focus:** Ensure ego vehicle departs before the NPC completes its overtaking maneuver because its velocity is too low.
- **Scenario Link:** 
	- [UC-NTR-001-0004_case1](https://evaluation.ci.tier4.jp/evaluation/scenarios/85d6da59-ffe8-463e-ae1b-3edef764f79b?project_id=awf)
	- [UC-NTR-001-0004_case2](https://evaluation.ci.tier4.jp/evaluation/scenarios/7f7338a3-e89d-4baa-8c1d-f8492795db45?project_id=awf)
- **Map Link:** [LEO-VM-00009](https://evaluation.ci.tier4.jp/evaluation/maps/980?project_id=awf)
- **Issue/PR Links:**
</details>

<details>
<summary>UC-NTR-001-0005</summary>

- **Scenario Description:** Ego vehicle is overtaken from the opposite direction lane by a faster moving NPC before moving.
- **Development Focus:** Ensure ego vehicle allows the high velocity NPC to overtake.
- **Scenario Link:** [UC-NTR-001-0005](https://evaluation.ci.tier4.jp/evaluation/scenarios/2edfabd2-d2ef-4ae1-8d37-58d7dceb6576?project_id=awf)
- **Map Link:** [LEO-VM-00010](https://evaluation.ci.tier4.jp/evaluation/maps/981?project_id=awf)
- **Issue/PR Links:**
</details>

<details>
<summary>UC-NTR-001-0006</summary>

- **Scenario Description:** Ego vehicle is overtaken from the opposite direction lane by a slower moving NPC before moving.
- **Development Focus:** Ensure ego vehicle departs before the NPC completes its overtaking maneuver because its velocity is too low..
- **Scenario Link:** [UC-NTR-001-0006](https://evaluation.ci.tier4.jp/evaluation/scenarios/15396f83-daa6-4eb9-9c7c-618050a924a3?project_id=awf)
- **Map Link:** [LEO-VM-00010](https://evaluation.ci.tier4.jp/evaluation/maps/981?project_id=awf)
- **Issue/PR Links:**
</details>

<details>
<summary>UC-NTR-001-0007</summary>

- **Scenario Description:** Ego vehicle is overtaken from the same direction lane by a faster moving NPC while driving.
- **Development Focus:** Ensure ego vehicle maintains safe distance and trajectory during overtaking.
- **Scenario Link:** [UC-NTR-001-0007](https://evaluation.ci.tier4.jp/evaluation/scenarios/83ef39b2-c573-485f-bb17-bd595f626d24?project_id=awf)
- **Map Link:** [LEO-VM-00009](https://evaluation.ci.tier4.jp/evaluation/maps/980?project_id=awf)
- **Issue/PR Links:**
</details>

<details>
<summary>UC-NTR-001-0008</summary>

- **Scenario Description:** Ego vehicle is overtaken from the same direction lane by a faster moving NPC which then slows down.
- **Development Focus:** Ensure ego vehicle adapts to changes in NPC's speed without significant deceleration.
- **Scenario Link:** [UC-NTR-001-0008](https://evaluation.ci.tier4.jp/evaluation/scenarios/cc6b1c53-61bd-444a-bced-10231d0cdbe7?project_id=awf)
- **Map Link:** [LEO-VM-00009](https://evaluation.ci.tier4.jp/evaluation/maps/980?project_id=awf)
- **Issue/PR Links:**
</details>

<details>
<summary>UC-NTR-002-0001</summary>

- **Scenario Description:** Ego vehicle overtakes a stopping or slow-moving NPC on the side of the road.
- **Development Focus:** Ensure safe overtaking of stationary or slow-moving obstacles.
- **Scenario Links:** 
	- [UC-NTR-002-0001_case1](https://evaluation.ci.tier4.jp/evaluation/scenarios/d3520449-a045-4787-86e0-bb243014b117?project_id=awf)
	- [UC-NTR-002-0001_case2](https://evaluation.ci.tier4.jp/evaluation/scenarios/577641a4-c755-4db1-aa6c-1dcadc0ada60?project_id=awf)
- **Map Link:** [LEO-VM-0008](https://evaluation.ci.tier4.jp/evaluation/maps/979?project_id=awf)
- **Issue/PR Links:**
</details>

<details>
<summary>UC-NTR-002-0002</summary>

- **Scenario Description:** Ego vehicle overtakes NPCs parked on different sides of the road.
- **Development Focus:** Navigate and maneuver around multiple parked obstacles safely.
- **Scenario Links:** 
	- [UC-NTR-002-0002_case1](https://evaluation.ci.tier4.jp/evaluation/scenarios/c4ee37cd-e1c9-487f-ba15-3d07db8c87c4?project_id=awf)
	- [UC-NTR-002-0002_case2](https://evaluation.ci.tier4.jp/evaluation/scenarios/8992be0b-22a3-4a61-ade9-fdf3e07de9b2?project_id=awf)
- **Map Link:** [LEO-VM-0008](https://evaluation.ci.tier4.jp/evaluation/maps/979?project_id=awf)
- **Issue/PR Links:**
</details>

<details>
<summary>UC-NTR-002-0003</summary>

- **Scenario Description:** Ego vehicle overtakes moving NPCs on the side of the road.
- **Development Focus:** Ensure dynamic decision-making for overtaking moving obstacles.
- **Scenario Links:** 
	- [UC-NTR-002-0003_case1](https://evaluation.ci.tier4.jp/evaluation/scenarios/f1c4b217-e246-4e95-90c1-037e14583365?project_id=awf)
	- [UC-NTR-002-0003_case2](https://evaluation.ci.tier4.jp/evaluation/scenarios/4236876f-7f1a-440b-8edd-eb83bee97f47?project_id=awf)
- **Map Link:** [LEO-VM-0008](https://evaluation.ci.tier4.jp/evaluation/maps/979?project_id=awf)
- **Issue/PR Links:**
</details>

<details>
<summary>UC-NTR-002-0004</summary>

- **Scenario Description:** Npc is overtaking another Npc in opposite side. It suddenly appears on Ego's trajectory.
- **Development Focus:** Ensure Ego can slowdown for vehicle comes from opposite direction.
- **Scenario Link:** [UC-NTR-002-0004](https://evaluation.ci.tier4.jp/evaluation/scenarios/6a86e217-8ada-4f54-9ded-1d77648b4257?project_id=awf)
- **Map Link:** [LEO-VM-0010](https://evaluation.ci.tier4.jp/evaluation/maps/981?project_id=awf)
- **Issue/PR Links:**
</details>

<details>
<summary>UC-NTR-003-0001</summary>

- **Scenario Description:** Ego vehicle decides to stop or pass in an intersection while NPC interferes with its trajectory.
- **Development Focus:** Handle intersection if intersection module is not triggered.
- **Scenario Link:** [UC-NTR-003-0001](https://evaluation.ci.tier4.jp/evaluation/scenarios/037f2b6b-0586-4d1c-9d4e-c43aee394a99?project_id=awf)
- **Map Link:** [Nishi-Shinjuku](https://evaluation.ci.tier4.jp/evaluation/maps/1256?project_id=awf)
- **Issue/PR Links:**
</details>

<details>
<summary>UC-NTR-003-0002</summary>

- **Scenario Description:** Ego vehicle decides to stop or pass in an intersection while NPC interferes with its trajectory during a turn.
- **Development Focus:** Ensure safe intersection handling during turning maneuvers. Ensure there won't be any unnecessary stopping.
- **Scenario Link:** [UC-NTR-003-0002](https://evaluation.ci.tier4.jp/evaluation/scenarios/1ac5d90e-aafa-420e-8aff-692a9cc3e039?project_id=awf)
- **Map Link:** [Nishi-Shinjuku](https://evaluation.ci.tier4.jp/evaluation/maps/1256?project_id=awf)
- **Issue/PR Links:**
</details>

<details>
<summary>UC-NTR-003-0003</summary>

- **Scenario Description:** Ego vehicle decides to stop or pass in an intersection while NPC interferes with its trajectory from the right.
- **Development Focus:** Ensure intersection module is working for low velocity road users.
- **Scenario Link:** [UC-NTR-003-0003](https://evaluation.ci.tier4.jp/evaluation/scenarios/80bf640c-702d-4c36-bc7d-f12a41f892bf?project_id=awf)
- **Map Link:** [awf_cicd_virtual_E_dev](https://evaluation.ci.tier4.jp/evaluation/maps/908?project_id=awf)
- **Issue/PR Links:**
</details>

<details>
<summary>UC-NTR-004-0001</summary>

- **Scenario Description:** Ego vehicle navigates a bidirectional lane with no line markings.
- **Development Focus:** Ensure safe navigation in undivided lanes with oncoming traffic.
- **Scenario Link:** [UC-NTR-004-0001](https://evaluation.ci.tier4.jp/evaluation/scenarios/446e8572-cbae-489c-97de-782cef10ed6d?project_id=awf)
- **Map Link:** [LEO-VM-00038](https://evaluation.ci.tier4.jp/evaluation/maps/1328?project_id=awf)
- **Issue/PR Links:**
</details>

<details>
<summary>UC-NTR-005-0001</summary>

- **Scenario Description:** Ego vehicle cruises behind a front vehicle.
- **Development Focus:** Maintain safe following distance and speed adjustment.
- **Scenario Link:** [UC-NTR-005-0001](https://evaluation.ci.tier4.jp/evaluation/scenarios/b2bfd6b6-f6ab-40c8-b249-231ca3cc7391?project_id=awf)
- **Map Link:** [LEO-VM-00001](https://evaluation.ci.tier4.jp/evaluation/maps/972?project_id=awf)
- **Issue/PR Links:**
</details>

<details>
<summary>UC-VRU-001-0001</summary>

- **Scenario Description:** Ego vehicle passes over a group of pedestrians.
- **Development Focus:** Navigate pedestrian dense areas without collisions.
- **Scenario Links:** 
	- [UC-VRU-001-0001_case1](https://evaluation.ci.tier4.jp/evaluation/scenarios/d1fb732c-d8e2-4708-83b0-ff2791c6d971?project_id=awf)
	- [UC-VRU-001-0001_case2](https://evaluation.ci.tier4.jp/evaluation/scenarios/5359eb4c-fe6b-4ad1-981a-cbe99bc3a1f6?project_id=awf)
- **Map Link:** [LEO-VM-0008](https://evaluation.ci.tier4.jp/evaluation/maps/979?project_id=awf)
- **Issue/PR Links:**
</details>

<details>
<summary>UC-VRU-001-0002</summary>

- **Scenario Description:** Ego vehicle slows down for an unexpected pedestrian.
- **Development Focus:** React to sudden pedestrian crossings.
- **Scenario Link:** [UC-VRU-001-0002](https://evaluation.ci.tier4.jp/evaluation/scenarios/236739b9-76fe-42bf-9121-d90deeaf99cc?project_id=awf)
- **Map Link:** [LEO-VM-00010](https://evaluation.ci.tier4.jp/evaluation/maps/981?project_id=awf)
- **Issue/PR Links:**
</details>

<details>
<summary>UC-VRU-002-0001</summary>

- **Scenario Description:** Ego vehicle allows a bicycle to drive first.
- **Development Focus:** Ensure safe interaction with bicycles on the road.
- **Scenario Links:** 
	- [UC-VRU-002-0001_case1](https://evaluation.ci.tier4.jp/evaluation/scenarios/9dff2eb1-7cfa-4b82-872f-0e3b138c03a2?project_id=awf)
	- [UC-VRU-002-0001_case2](https://evaluation.ci.tier4.jp/evaluation/scenarios/4af5839f-477d-450b-92db-7aefa0deceff?project_id=awf)
- **Map Link:** [LEO-VM-0008](https://evaluation.ci.tier4.jp/evaluation/maps/979?project_id=awf)
- **Issue/PR Links:**
</details>

<details>
<summary>UC-VRU-002-0002</summary>

- **Scenario Description:** Ego vehicle drives through destination while a bicycle is present.
- **Development Focus:** Manage overtaking and following scenarios involving bicycles.
- **Scenario Links:** 
	- [UC-VRU-002-0002_case1](https://evaluation.ci.tier4.jp/evaluation/scenarios/9dff2eb1-7cfa-4b82-872f-0e3b138c03a2?project_id=awf)
	- [UC-VRU-002-0002_case2](https://evaluation.ci.tier4.jp/evaluation/scenarios/4af5839f-477d-450b-92db-7aefa0deceff?project_id=awf)
- **Map Link:** [LEO-VM-0001](https://evaluation.ci.tier4.jp/evaluation/maps/972?project_id=awf)
- **Issue/PR Links:**
</details>

<details>
<summary>UC-VRU-002-0003</summary>

- **Scenario Description:** Ego vehicle drives through destination while a bicycle is driving on the side of the road.
- **Development Focus:** Ensure smooth interaction with driving bicycles while avoiding it.
- **Scenario Links:** 
	- [UC-VRU-002-0003_case1](https://evaluation.ci.tier4.jp/evaluation/scenarios/e4616159-5531-4dfe-928b-c2039d756020?project_id=awf)
	- [UC-VRU-002-0003_case2](https://evaluation.ci.tier4.jp/evaluation/scenarios/e4616159-5531-4dfe-928b-c2039d756020?project_id=awf)
- **Map Link:** [LEO-VM-0008](https://evaluation.ci.tier4.jp/evaluation/maps/979?project_id=awf)
- **Issue/PR Links:**
</details>

<details>
<summary>UC-VRU-002-0004</summary>

- **Scenario Description:** Ego vehicle drives through destination while a bicycle is approaching from outside the road.
- **Development Focus:** Decelerate comfortably to avoid collisions.
- **Scenario Link:** 
	- [UC-VRU-002-0004_case1](https://evaluation.ci.tier4.jp/evaluation/scenarios/a56881a3-8cbd-4fbb-9dbd-c28719f7fe9b?project_id=awf)
	- [UC-VRU-002-0004_case2](https://evaluation.ci.tier4.jp/evaluation/scenarios/3fa16833-02e8-4b60-b2d6-33aac6b67339?project_id=awf)
- **Map Link:** [LEO-VM-0008](https://evaluation.ci.tier4.jp/evaluation/maps/979?project_id=awf)
- **Issue/PR Links:**
</details>

<details>
<summary>UC-VRU-002-0005</summary>

- **Scenario Description:** Ego vehicle drives through the destination while two bicycles drive in the same direction.
- **Development Focus:** Wait for NPCs to pass before overtaking.
- **Scenario Link:** [UC-VRU-002-0005](https://evaluation.ci.tier4.jp/evaluation/scenarios/19f99f99-ac8e-4b55-8f3e-cddadd7f349b?project_id=awf)
- **Map Link:** [LEO-VM-00038](https://evaluation.ci.tier4.jp/evaluation/maps/1328?project_id=awf)
- **Issue/PR Links:**
</details>

</details>

### Purpose

The primary goal of this task is to enhance the capability of autonomous driving software for low-speed driving in dense urban areas with mixed road users. This involves improving the system's awareness of various conditions and its responsiveness to these conditions. By fine-tuning trajectory planning, cruise control for dynamic obstacles, adaptive cruise planning, and avoidance strategies, we aim to ensure smoother, safer, and more efficient driving in complex urban environments.

### Possible approaches

| Improvement Target                      | Description                                                                                     | Related Scenarios    | Assignees               |
|-----------------------------------------|-------------------------------------------------------------------------------------------------|----------------------|-------------------------|
| **Path Planning and Trajectory Adjustment**    | - Adjust the trajectory to align more closely with the right or left edge of the lane for bi-directional lanes, depending on the direction of traffic flow. | <br> - UC-VRU-002-0005 <br> - UC-NTR-004-0001           | [Name]             |
| **Cruise Planning for Dynamic Obstacles**      | - Expand the coverage area to consider objects further away from the trajectory. <br> - Autoware should react much earlier if the predicted paths of other vehicles intersect with our planned path to avoid potential collisions. <br> - Avoid unnecessary reactions to objects that briefly enter and exit our path to prevent abrupt and unnecessary slowdowns, ensuring smoother and more efficient driving. | <br> - UC-NTR-001-0001 <br> - UC-NTR-001-0002 <br> - UC-NTR-001-0003 <br> - UC-NTR-001-0004 <br> - UC-NTR-001-0005 <br> - UC-NTR-001-0006 <br> - UC-NTR-002-0004 <br> - UC-NTR-003-0001 <br> - UC-NTR-003-0002 <br> - UC-NTR-003-0003 <br> - UC-VRU-001-0002 <br> - UC-VRU-002-0001 <br> - UC-VRU-002-0004          | <br> - @brkay54             |
| **Adaptive Cruise Planning for Following Obstacles** | - The ego vehicle may overshoot its velocity while following a dynamic object; find a better approach for velocity planning while following an object. <br> - Maintain a following distance within 2-3 seconds. | <br> - UC-NTR-001-0007 <br> - UC-NTR-001-0008 <br> - UC-NTR-005-0001 <br> - UC-VRU-002-0002          | <br> - @ahmeddesokyebrahim              |
| **Static and Dynamic Avoidance**              | - The system needs better capabilities to dynamically avoid objects coming from the opposite direction. Test and improve dynamic avoidance to avoid oncoming objects based on their future positions. <br> - Test current avoidance and develop enhanced safe avoidance capabilities for vulnerable road users. | <br> - UC-NTR-002-0001 <br> - UC-NTR-002-0002 <br> - UC-NTR-002-0003  <br> - UC-NTR-002-0004 <br> - UC-NTR-004-0001 <br> - UC-VRU-001-0001 <br> - UC-VRU-002-0003     | <br> - @beyzanurkaya  <br> - @sasakisasaki             |

**NOTE:** A scenario might be in multiple improvement targets because the scenario needs both improvements to pass the evaluation.

### Definition of done

- [ ] **Problem Identification and Solution Proposal:** Developers have created issues that clearly define the problem and propose a viable solution.
- [ ] **Implementation of Improvements:** All necessary improvements have been made to the relevant software packages.
- [ ] **Addition of New Scenarios:** Additional scenarios have been added into Autoware Evaluator to test other various conditions.
- [ ] **Successful Scenario Execution:** All scenarios have been executed successfully, with the system passing each one without issues.

**Comment by mitsudome-r** [comment]
https://docs.google.com/spreadsheets/d/1WTqiyUndtdBNgiNnAApjOXhma-QAU_SBPdD626pMr2Q/edit?gid=0#gid=0
